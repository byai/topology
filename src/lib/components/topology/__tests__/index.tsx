/* eslint-disable no-undef */
import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import ReactDOM from 'react-dom';
import 'jest-dom/extend-expect';
import { wrapInTestContext } from 'react-dnd-test-utils';
import TestRenderer from 'react-test-renderer';
// eslint-disable-next-line import/no-named-as-default, import/no-named-as-default-member
import Topology from '..';

afterEach(cleanup);

describe('topology', () => {
    let container;
    const BoxContext = wrapInTestContext(Topology);
    const renderTreeNode = data => (<div style={{ width: 100, height: 100 }}>{data.id}</div>);

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    afterEach(() => {
        document.body.removeChild(container);
        container = null;
    });

    test('basic render', () => {
        const mockRenderTreeNode = jest.fn(data => (<div>{data.id}</div>));
        const { baseElement } = render(
            <BoxContext
                data={{ nodes: [{ id: '1' }], lines: [] }}
                onChange={() => null}
                renderTreeNode={mockRenderTreeNode}
            />,
        );
        expect(mockRenderTreeNode.mock.calls.length).toBeGreaterThan(0);
        expect(baseElement).toMatchSnapshot();
    });

    test('autoLayout', () => {
        const mockRenderTreeNode = jest.fn(data => (<div>{data.id}</div>));
        const mockChange = jest.fn(() => null);
        let topologyInstance = null;
        TestRenderer.act(
            () => {
                ReactDOM.render(
                    <BoxContext
                        getInstance={(r) => { topologyInstance = r; }}
                        data={{
                            nodes: [{ id: '1' }, { id: '2' }],
                            lines: [{ start: '1-0', end: '2' }],
                        }}
                        onChange={mockChange}
                        renderTreeNode={mockRenderTreeNode}
                    />,
                    container,
                );
            },
        );
        expect(topologyInstance === null).toBeFalsy();
        topologyInstance.autoLayout();
        expect(mockChange.mock.calls.length).toBeGreaterThan(0);
    });

    test('scrollCanvasToCenter', () => {
        let topologyInstance = null;
        TestRenderer.act(
            () => {
                ReactDOM.render(
                    <BoxContext
                        getInstance={(r) => { topologyInstance = r; }}
                        data={{ nodes: [{ id: '1', position: { x: 2000, y: 2000 } }], lines: [] }}
                        onChange={() => null}
                        renderTreeNode={renderTreeNode}
                    />,
                    container,
                );
            },
        );
        topologyInstance.$wrapper.scrollLeft = 0;
        topologyInstance.$wrapper.scrollTop = 0;
        expect(topologyInstance === null).toBeFalsy();
        const scrollBtn = container.querySelector('#scroll-canvas-to-center');
        fireEvent(scrollBtn, new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
        }));
        expect(topologyInstance.$wrapper.scrollLeft).toBeGreaterThan(0);
        expect(topologyInstance.$wrapper.scrollTop).toBeGreaterThan(0);
    });
});
