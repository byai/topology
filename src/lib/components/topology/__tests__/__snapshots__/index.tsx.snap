// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topology basic render 1`] = `
<body>
  <div />
  <div>
    <div
      class="byai-topology"
    >
      <div
        class="topology-wrapper"
      >
        <div
          class="topology-canvas topology-zoom"
          id="topology-canvas"
          style="width: 40000px; height: 40000px;"
        >
          <div
            class="byai-topology-node-wrapper"
            id="topology-node-1"
            style="position: absolute; left: 20000px; top: 20000px; transition: .2s; z-index: 999;"
          >
            <div
              class="topology-node-preview"
              data-id="1"
            />
            <div
              class="topology-node-content"
            >
              <div>
                1
              </div>
            </div>
          </div>
          <svg
            class="topology-svg"
          />
          <div
            class="byai-topology-selection"
            style="left: 0px; top: 0px; width: 0px; height: 0px;"
          />
          <svg
            class="snap-line-svg"
          />
        </div>
      </div>
      <div
        class="topology-tools"
        data-html2canvas-ignore="false"
      >
        <div
          class="topology-tools-btn"
          id="scroll-canvas-to-center"
        >
          <img
            alt=""
            src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBzdGFuZGFsb25lPSJubyI/PjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2ZzExLmR0ZCI+PHN2ZyB0PSIxNTYwMzI3NjY4NDk5IiBjbGFzcz0iaWNvbiIgc3R5bGU9IiIgdmlld0JveD0iMCAwIDEwMjQgMTAyNCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHAtaWQ9IjExMzciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCI+PGRlZnM+PHN0eWxlIHR5cGU9InRleHQvY3NzIj48L3N0eWxlPjwvZGVmcz48cGF0aCBkPSJNNTEzLjc5OTY0OSAyOTUuNzQyMjM4Yy0xMTguMTc2OTE5IDAtMjE0LjE1ODE3MiA5Ni4xODEyMTUtMjE0LjE1ODE3MyAyMTQuMTU4MTcyIDAgMTE4LjE3NjkxOSA5Ni4xODEyMTUgMjE0LjE1ODE3MiAyMTQuMTU4MTczIDIxNC4xNTgxNzIgMTE3Ljk3Njk1OCAwIDIxNC4xNTgxNzItOTUuNzgxMjkzIDIxNC4xNTgxNzItMjEzLjk1ODIxMXMtOTYuMTgxMjE1LTIxNC4zNTgxMzMtMjE0LjE1ODE3Mi0yMTQuMzU4MTMzeiBtMCAzNjcuMzI4MjU2Yy04NC4zODM1MTkgMC0xNTIuOTcwMTIzLTY4LjU4NjYwNC0xNTIuOTcwMTI0LTE1Mi45NzAxMjNzNjguNTg2NjA0LTE1Mi45NzAxMjMgMTUyLjk3MDEyNC0xNTIuOTcwMTIzIDE1Mi45NzAxMjMgNjguNTg2NjA0IDE1Mi45NzAxMjMgMTUyLjk3MDEyMy02OC41ODY2MDQgMTUyLjk3MDEyMy0xNTIuOTcwMTIzIDE1Mi45NzAxMjN6IiBmaWxsPSIjMzMzMzMzIiBwLWlkPSIxMTM4Ij48L3BhdGg+PHBhdGggZD0iTTk5MS4zMDYzODUgNDgwLjUwNjE1MUg5MTMuOTIxNWMtNy4xOTg1OTQtOTYuNTgxMTM2LTQ4LjE5MDU4OC0xODYuMzYzNjAxLTExNy4zNzcwNzUtMjU1LjU1MDA4OHMtMTU4Ljk2ODk1MS0xMTAuMTc4NDgxLTI1NS41NTAwODgtMTE3LjM3NzA3NVYzMC41OTQwMjVjMC0xNi43OTY3MTktMTMuNzk3MzA1LTMwLjU5NDAyNS0zMC41OTQwMjUtMzAuNTk0MDI1cy0zMC41OTQwMjUgMTMuNzk3MzA1LTMwLjU5NDAyNCAzMC41OTQwMjV2NzYuOTg0OTYzYy05Ni4zODExNzYgNy4xOTg1OTQtMTg2LjM2MzYwMSA0OC4xOTA1ODgtMjU1LjU1MDA4OCAxMTcuMzc3MDc1UzExNC4wNzc3MTkgMzgzLjcyNTA1NCAxMDYuODc5MTI1IDQ4MC41MDYxNTFIMzAuNjk0MDA1Yy0xNi43OTY3MTkgMC0zMC41OTQwMjUgMTMuNzk3MzA1LTMwLjU5NDAyNSAzMC41OTQwMjVzMTMuNzk3MzA1IDMwLjU5NDAyNSAzMC41OTQwMjUgMzAuNTk0MDI0aDc2LjE4NTEyYzcuMTk4NTk0IDk2LjU4MTEzNiA0OC4xOTA1ODggMTg2LjM2MzYwMSAxMTcuMzc3MDc1IDI1NS41NTAwODhzMTU4Ljk2ODk1MSAxMTAuMTc4NDgxIDI1NS41NTAwODggMTE3LjM3NzA3NXY3OC43ODQ2MTJjMCAxNi43OTY3MTkgMTMuNzk3MzA1IDMwLjU5NDAyNSAzMC41OTQwMjQgMzAuNTk0MDI1czMwLjU5NDAyNS0xMy43OTczMDUgMzAuNTk0MDI1LTMwLjU5NDAyNXYtNzguNzg0NjEyYzk2LjM4MTE3Ni03LjE5ODU5NCAxODYuMzYzNjAxLTQ4LjE5MDU4OCAyNTUuNTUwMDg4LTExNy4zNzcwNzVzMTEwLjE3ODQ4MS0xNTguOTY4OTUxIDExNy4zNzcwNzUtMjU1LjU1MDA4OGg3Ny4zODQ4ODVjMTYuNzk2NzE5IDAgMzAuNTk0MDI1LTEzLjc5NzMwNSAzMC41OTQwMjUtMzAuNTk0MDI0IDAtMTYuOTk2NjgtMTMuNzk3MzA1LTMwLjU5NDAyNS0zMC41OTQwMjUtMzAuNTk0MDI1ek03NTMuMTUyOSA3NTMuODUyNzYzYy02NC43ODczNDYgNjQuNzg3MzQ2LTE1MS4xNzA0NzUgMTAwLjU4MDM1NS0yNDIuNzUyNTg4IDEwMC41ODAzNTYtOTEuNzgyMDc0IDAtMTc3Ljk2NTI0MS0zNS43OTMwMDktMjQyLjc1MjU4Ny0xMDAuNTgwMzU2LTY0Ljc4NzM0Ni02NC43ODczNDYtMTAwLjU4MDM1NS0xNTEuMTcwNDc1LTEwMC41ODAzNTUtMjQyLjc1MjU4N3MzNS43OTMwMDktMTc3Ljk2NTI0MSAxMDAuNTgwMzU1LTI0Mi43NTI1ODhjNjQuNzg3MzQ2LTY0Ljc4NzM0NiAxNTEuMTcwNDc1LTEwMC41ODAzNTUgMjQyLjc1MjU4Ny0xMDAuNTgwMzU1IDkxLjc4MjA3NCAwIDE3Ny45NjUyNDEgMzUuNzkzMDA5IDI0Mi43NTI1ODggMTAwLjU4MDM1NSA2NC43ODczNDYgNjQuNzg3MzQ2IDEwMC41ODAzNTUgMTUxLjE3MDQ3NSAxMDAuNTgwMzU1IDI0Mi43NTI1ODhzLTM1LjU5MzA0OCAxNzcuOTY1MjQxLTEwMC41ODAzNTUgMjQyLjc1MjU4N3oiIGZpbGw9IiMzMzMzMzMiIHAtaWQ9IjExMzkiPjwvcGF0aD48L3N2Zz4="
          />
          <div
            class="tooltip"
          >
            定位中心
          </div>
        </div>
        <div
          class="topology-tools-btn"
          id="auto-layout"
        >
          <img
            alt=""
            src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBzdGFuZGFsb25lPSJubyI/PjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2ZzExLmR0ZCI+PHN2ZyB0PSIxNTYwMzI3NjU0MDc4IiBjbGFzcz0iaWNvbiIgc3R5bGU9IiIgdmlld0JveD0iMCAwIDEwMjQgMTAyNCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHAtaWQ9IjEwMjIiIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCI+PGRlZnM+PHN0eWxlIHR5cGU9InRleHQvY3NzIj48L3N0eWxlPjwvZGVmcz48cGF0aCBkPSJNODg1LjkyNzYxNyA2NzQuMzAwMTMyVjQ5Ny42NDYxMTNINTQwLjc3MTIyVjM1My44Mjk5MjRoMzQ1LjE1NzQyMVYxMjMuNjk3MDA2SDEzOC4wNzAzMzZ2MjMwLjEzMjkxOGgzNDUuMTU3NDIxdjE0My44MTYxODlIMTM4LjA3MDMzNnYxNzYuNjU0MDE5Yy00OS40Nzc3NzkgMTIuODYyOTMzLTg2LjI4ODA3NiA1Ny40OTc0MTQtODYuMjg4MDc2IDExMC45NTA3MjkgMCA2My40NTMwNDQgNTEuNTk4MDY1IDExNS4wNTIxMzMgMTE1LjA1NDE3OSAxMTUuMDUyMTMzIDYzLjQ1MjAyMSAwIDExNS4wNTAwODYtNTEuNTk5MDg5IDExNS4wNTAwODYtMTE1LjA1MjEzMyAwLTUzLjQ1MzMxNi0zNi44MTAyOTctOTguMDg3Nzk2LTg2LjI4ODA3Ni0xMTAuOTUwNzI5VjU1NS4xNDQ1NWgyODcuNjMwMzMxdjExOS4xNTQ1NTljLTQ5LjQ5MTA4MiAxMi44NjI5MzMtODYuMjg4MDc2IDU3LjQ5NzQxNC04Ni4yODgwNzYgMTEwLjk1MDcyOSAwIDYzLjQ1MzA0NCA1MS41OTkwODkgMTE1LjA1MjEzMyAxMTUuMDY2NDU5IDExNS4wNTIxMzMgNjMuNDI1NDE1IDAgMTE1LjA1NDE3OS01MS41OTkwODkgMTE1LjA1NDE3OS0xMTUuMDUyMTMzIDAtNTMuNDUzMzE2LTM2Ljc5ODAxNy05OC4wODc3OTYtODYuMjg5MDk5LTExMC45NTA3MjlWNTU1LjE0NDU1aDI4Ny42MzAzMzF2MTE5LjE1NDU1OWMtNDkuNDkzMTI5IDEyLjg2MjkzMy04Ni4yODcwNTMgNTcuNDk3NDE0LTg2LjI4NzA1MyAxMTAuOTUwNzI5IDAgNjMuNDUzMDQ0IDUxLjU2OTQxMyAxMTUuMDUyMTMzIDExNS4wNTExMSAxMTUuMDUyMTMzIDYzLjQyNDM5MSAwIDExNS4wNTMxNTYtNTEuNTk5MDg5IDExNS4wNTMxNTYtMTE1LjA1MjEzMy0wLjAwMjA0Ny01My40NTMzMTYtMzYuODAwMDY0LTk4LjA4Njc3My04Ni4yOTIxNy0xMTAuOTQ5NzA2ek0xOTUuNTk3NDI2IDE4MS4yMjQwOTZoNjMyLjgwNDEyNXYxMTUuMDUyMTMySDE5NS41OTc0MjZWMTgxLjIyNDA5NnogbTI4Ljc2NDA1NiA2MDQuMDI2NzY1YzAgMzEuNzEyMTk2LTI1LjgxNDg5NCA1Ny41NTQ3MTktNTcuNTI2MDY2IDU3LjU1NDcxOS0zMS43MTQyNDIgMC01Ny41MjcwOS0yNS44NDI1MjMtNTcuNTI3MDktNTcuNTU0NzE5IDAtMzEuNzE0MjQyIDI1LjgxMjg0Ny01Ny40OTg0MzcgNTcuNTI3MDktNTcuNDk4NDM3IDMxLjcxMjE5Ni0wLjAwMTAyMyA1Ny41MjYwNjYgMjUuNzgzMTcxIDU3LjUyNjA2NiA1Ny40OTg0Mzd6IG0zNDUuMTcxNzQ3IDBjMCAzMS43MTIxOTYtMjUuODQxNSA1Ny41NTQ3MTktNTcuNTI3MDg5IDU3LjU1NDcxOS0zMS43MjQ0NzUgMC01Ny41MzkzNjktMjUuODQyNTIzLTU3LjUzOTM2OS01Ny41NTQ3MTkgMC0zMS43MTQyNDIgMjUuODE1OTE3LTU3LjQ5ODQzNyA1Ny41MzkzNjktNTcuNDk4NDM3IDMxLjY4NTU5LTAuMDAxMDIzIDU3LjUyNzA5IDI1Ljc4MzE3MSA1Ny41MjcwODkgNTcuNDk4NDM3eiBtMjg3LjYzMTM1NSA1Ny41NTQ3MTljLTMxLjc0MTg3MiAwLTU3LjUyODExMy0yNS44NDI1MjMtNTcuNTI4MTEzLTU3LjU1NDcxOSAwLTMxLjcxNDI0MiAyNS43ODYyNDEtNTcuNDk4NDM3IDU3LjUyODExMy01Ny40OTg0MzcgMzEuNjg1NTkgMCA1Ny41MjcwOSAyNS43ODQxOTUgNTcuNTI3MDkgNTcuNDk4NDM3IDAgMzEuNzEyMTk2LTI1Ljg0MTUgNTcuNTU0NzE5LTU3LjUyNzA5IDU3LjU1NDcxOXoiIGZpbGw9IiMyYzJjMmMiIHAtaWQ9IjEwMjMiPjwvcGF0aD48L3N2Zz4="
          />
          <div
            class="tooltip"
          >
            自动布局
          </div>
        </div>
        <div
          class="topology-tools-zoom"
        >
          <img
            alt=""
            src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBzdGFuZGFsb25lPSJubyI/PjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2ZzExLmR0ZCI+PHN2ZyB0PSIxNjIxOTIzNzE5OTI0IiBjbGFzcz0iaWNvbiIgdmlld0JveD0iMCAwIDEwMjQgMTAyNCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHAtaWQ9IjEwOTIwIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgd2lkdGg9IjEyOCIgaGVpZ2h0PSIxMjgiPjxkZWZzPjxzdHlsZSB0eXBlPSJ0ZXh0L2NzcyI+PC9zdHlsZT48L2RlZnM+PHBhdGggZD0iTTUxMi42MTY5NDIgNjQuMTg2NjczYzYxLjcyMDU1MyAwIDExOS43OTA3ODIgMTEuNzc5ODEzIDE3NC4yMTEwODUgMzUuMzM5NDRzMTAxLjcwNjA0MyA1NS40MTU0MjggMTQxLjg1NzQyMSA5NS41Njc0MDVjNDAuMTUxMzc3IDQwLjE1MTk3NyA3Mi4wMDcxNzkgODcuMjcxODMgOTUuNTY3NDA1IDE0MS4zNTk3NTggMjMuNTYwMjI2IDU0LjA4NzcyOSAzNS4zNDAwMzkgMTExLjk5MjI2OSAzNS4zMzk0NCAxNzMuNzEzNDIyLTAuMDAwNiA2MS43MjA5NTMtMTEuNzgwMjEzIDExOS43OTEzODEtMzUuMzM5NDQgMTc0LjIxMTA4NS0yMy41NTkwMjcgNTQuNDE5NzAzLTU1LjQxNTAyOSAxMDEuNzA1NDQ0LTk1LjU2NzQwNSAxNDEuODU3NDIxLTQwLjE1MjU3NyA0MC4xNTE5NzctODcuNDM4MzE3IDcyLjAwNzc3OS0xNDEuODU3NDIxIDk1LjU2NzQwNS01NC40MTkxMDQgMjMuNTU5NjI3LTExMi40ODkzMzIgMzUuMzM5NDQtMTc0LjIxMTA4NSAzNS4zMzk0NHMtMTE5LjYyNjA5NC0xMS43Nzk4MTMtMTczLjcxMzQyMi0zNS4zMzk0NGMtNTQuMDg3MzI5LTIzLjU1OTYyNy0xMDEuMjA3MTgyLTU1LjQxNTQyOC0xNDEuMzU5NzU4LTk1LjU2NzQwNS00MC4xNTI1NzctNDAuMTUxOTc3LTcyLjAwODM3OC04Ny40Mzc3MTctOTUuNTY3NDA1LTE0MS44NTc0MjFzLTM1LjMzODg0LTExMi40ODk5MzItMzUuMzM5NDQtMTc0LjIxMTA4NWMtMC4wMDA2LTYxLjcyMTE1MyAxMS43NzkyMTQtMTE5LjYyNTQ5NCAzNS4zMzk0NC0xNzMuNzEzNDIyczU1LjQxNjAyOC0xMDEuMjA3NzgxIDk1LjU2NzQwNS0xNDEuMzU5NzU4IDg3LjI3MTIzLTcyLjAwNzc3OSAxNDEuMzU5NzU4LTk1LjU2NzQwNVM0NTAuODk2NTg4IDY0LjE4Njg3MyA1MTIuNjE2OTQyIDY0LjE4NjY3M3pNNzM0LjYxMDgzIDU3MC44OTEzMjhjMTkuOTA5NzAxIDAgMzcuODI4NTUyLTUuMTQzMzEzIDUzLjc1NjE1My0xNS40Mjk5MzkgMTUuOTI3NjAxLTEwLjI4NjYyNiAyMy44OTE0MDItMjUuNzE2NTY0IDIzLjg5MTQwMi00Ni4yODk4MTYgMC0xOS45MDk3MDEtNy45NjM4MDEtMzQuNjc2Mjg5LTIzLjg5MTQwMi00NC4yOTkzNjUtMTUuOTI3NjAxLTkuNjIzMDc2LTMzLjg0NjI1Mi0xNC40MzQ2MTMtNTMuNzU2MTUzLTE0LjQzNDgxM0wyOTQuNjA0MTU0IDQ1MC40MzczOTVjLTE5LjkwOTcwMSAwLTM4LjE2MDUyNyA0LjgxMTUzOC01NC43NTIyNzggMTQuNDM0ODEzLTE2LjU5MTc1MSA5LjYyMzA3Ni0yNC44ODc1MjYgMjQuMzg5NjY0LTI0Ljg4NzUyNiA0NC4yOTkzNjUgMCAyMC41NzMyNTEgOC4yOTU3NzUgMzYuMDAzMTkgMjQuODg3NTI2IDQ2LjI4OTgxNiAxNi41OTE3NTEgMTAuMjg2NjI2IDM0Ljg0MjM3NyAxNS40Mjk5MzkgNTQuNzUyMjc4IDE1LjQyOTkzOUw3MzQuNjEwODMgNTcwLjg5MTMyOCA3MzQuNjEwODMgNTcwLjg5MTMyOHoiIHAtaWQ9IjEwOTIxIiBmaWxsPSIjOUZBMkE4Ij48L3BhdGg+PC9zdmc+"
          />
        </div>
        <div
          class="topology-tools-percent"
        >
          100%
        </div>
        <div
          class="topology-tools-zoom"
        >
          <img
            alt=""
            src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBzdGFuZGFsb25lPSJubyI/PjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2ZzExLmR0ZCI+PHN2ZyB0PSIxNjIxOTIzNDA4MzQyIiBjbGFzcz0iaWNvbiIgdmlld0JveD0iMCAwIDEwMjQgMTAyNCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHAtaWQ9IjgwNSIgZGF0YS1zcG0tYW5jaG9yLWlkPSJhMzEzeC43NzgxMDY5LjAuaTIiIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iMTI4IiBoZWlnaHQ9IjEyOCI+PGRlZnM+PHN0eWxlIHR5cGU9InRleHQvY3NzIj48L3N0eWxlPjwvZGVmcz48cGF0aCBkPSJNNTE3LjEyIDQzLjkxOTM2Yy0yNTguNzM0MDggMC00NjguNDggMjA5Ljc0NTkyLTQ2OC40OCA0NjguNDhzMjA5Ljc0NTkyIDQ2OC40OCA0NjguNDggNDY4LjQ4IDQ2OC40OC0yMDkuNzQ1OTIgNDY4LjQ4LTQ2OC40OC0yMDkuNzQ1OTItNDY4LjQ4LTQ2OC40OC00NjguNDh6TTc0Mi4yMzg3MiA1NjUuNzZINTcwLjg4djE3MS43OTkwNGMwIDI5LjU2OC0yNC4xOTIgNTMuNzYtNTMuNzYgNTMuNzZzLTUzLjc2LTI0LjE5Mi01My43Ni01My43NlY1NjUuNzZoLTE3MS40MzgwOGMtMjkuNTY4IDAtNTMuNzYtMjQuMTkyLTUzLjc2LTUzLjc2czI0LjE5Mi01My43NiA1My43Ni01My43Nkg0NjMuMzZ2LTE3MC45OTc3NmMwLTI5LjU2OCAyNC4xOTItNTMuNzYgNTMuNzYtNTMuNzZzNTMuNzYgMjQuMTkyIDUzLjc2IDUzLjc2VjQ1OC4yNGgxNzEuMzU4NzJjMjkuNTY4IDAgNTMuNzYgMjQuMTkyIDUzLjc2IDUzLjc2cy0yNC4xOTIgNTMuNzYtNTMuNzYgNTMuNzZ6IiBwLWlkPSI4MDYiIGZpbGw9IiM5RkEyQTgiPjwvcGF0aD48L3N2Zz4="
          />
        </div>
      </div>
    </div>
  </div>
  <div
    id="topology-dom-map"
  >
    <div>
      <div
        class="dom-map-wrapper"
        id="dom-map-1"
      >
        <div>
          1
        </div>
      </div>
    </div>
  </div>
</body>
`;
